dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/GroupUtil.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.utils;
+
+import java.sql.SQLException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.CommunityService;
+import org.dspace.core.Context;
+import org.dspace.eperson.Group;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * A class which provides utility methods for Groups
+ */
+@Component
+public class GroupUtil {
+
+    private GroupUtil() {
+    }
+
+    /**
+     * UUID regex used in the collection regex
+     */
+    private static final String UUID_REGEX = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0" +
+            "-9a-fA-F]{12}";
+    /**
+     * Collection regex used to extract the ID
+     */
+    private static final String COLLECTION_REGEX = "COLLECTION_(" + UUID_REGEX + ")_.*?";
+
+    /**
+     * The community prefix: all groups which are specific to
+     * a community start with this.
+     */
+    private static final String COMMUNITY_PREFIX = "COMMUNITY_";
+
+    /**
+     * These are the possible community suffixes. All groups which are
+     * specific to a collection will end with one of these. The collection
+     * id should be between the prefix and the suffix.
+     * <p>
+     * Note: the order of these suffixes are important, see getCollectionRole()
+     */
+    private static final String[] COMMUNITY_SUFFIXES = {"_ADMIN"};
```

Comment:
Tiny thing. The JavaDocs here are a bit wrong, as they talk about Collection instead of Community.  You could likely remove most of the Javadocs here.

Also, honestly, you could switch the Community Group checks to be the same as the Collection ones, therefore removing this `COMMUNITY_SUFFIXES` altogether.  People shouldn't be naming groups with a long UUID within them, and if they are, they likely want it related to the Community/Collection anyhow.  This isn't required of this PR, but just wanted to suggest it.

Reply:
The JavaDocs have been adjusted (and the `/subgroups` change was also completed)